### ISSUE TITLE
SemanticAnalyzer fails for incomplete match pattern clause

### ISSUE THREAD

**Issue by nadeeshaan** [ballerina-platform/ballerina-lang#25155]
**Description:**
Consider the following source snippet,

```
function bar (string | int | boolean i)  returns (string | int | boolean) {
    string | int | boolean var1 =  "Test";

    match var1 {
        v
    }

    return 1;
}

type Person record {
    string name;
    int age;
};

```

For the above, semantic analyzer fails with the following NPE,

```
org.ballerinalang.langserver.compiler.exception.CompilationFailedException: Oh no, something really went wrong. Bad. Sad.
	at org.ballerinalang.langserver.compiler.LSModuleCompiler.compilePackagesSafe(LSModuleCompiler.java:305)
	at org.ballerinalang.langserver.compiler.LSModuleCompiler.getBLangPackages(LSModuleCompiler.java:166)
	at org.ballerinalang.langserver.diagnostic.DiagnosticsHelper.compileAndSendDiagnostics(DiagnosticsHelper.java:76)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$didChange$16(BallerinaTextDocumentService.java:761)
	at org.ballerinalang.langserver.util.Debouncer.lambda$call$0(Debouncer.java:42)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.checkStaticMatchPatternLiteralType(SemanticAnalyzer.java:2317)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:2239)
	at org.wso2.ballerinalang.compiler.tree.statements.BLangMatch$BLangMatchStaticBindingPatternClause.accept(BLangMatch.java:179)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.lambda$visit$31(SemanticAnalyzer.java:2232)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:2230)
	at org.wso2.ballerinalang.compiler.tree.statements.BLangMatch.accept(BLangMatch.java:86)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:2610)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:2578)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeStmt(SemanticAnalyzer.java:2574)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:361)
	at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:57)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:2610)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:333)
	at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:73)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:2610)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:2578)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:2570)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:256)
	at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:161)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyze(SemanticAnalyzer.java:230)
	at org.wso2.ballerinalang.compiler.CompilerDriver.typeCheck(CompilerDriver.java:322)
	at org.wso2.ballerinalang.compiler.CompilerDriver.compile(CompilerDriver.java:256)
	at org.wso2.ballerinalang.compiler.CompilerDriver.compilePackageSymbol(CompilerDriver.java:248)
	at org.wso2.ballerinalang.compiler.CompilerDriver.compilePackage(CompilerDriver.java:140)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:166)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:137)
	at org.ballerinalang.langserver.compiler.LSModuleCompiler.compilePackagesSafe(LSModuleCompiler.java:291)
	... 11 more
```

**Comment by pubudu91** [comment]
@KavinduZoysa is rewriting the match and related stuff. Kavindu can you see if this gets fixed from your PR?

**Comment by KavinduZoysa** [comment]
This NPE is fixed in #25082

**Comment by SupunS** [comment]
@KavinduZoysa Do we have any tests covering this scenario in PR 25082?

**Comment by KavinduZoysa** [comment]
Added test cases to cover this scenario in #25082

**Comment by MaryamZi** [comment]
Closing this issue since it has been fixed in #25082 and tests have been added in 864c692.
